Spurious pytest failure

### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.


### Issue description

I believe that the the parallelization lead to a race condition with a file. 

@stinodego I think we must ensure that all file creators and consumers end up on the same workers.

```[gw4] linux -- Python 3.9.12 /home/ritchie46/code/polars/py-polars/venv/bin/python

    @pytest.mark.xfail(sys.platform == "win32", reason="Does not work on Windows")
    def test_parquet_struct_categorical() -> None:
        df = pl.DataFrame(
            [
                pl.Series("a", ["bob"], pl.Categorical),
                pl.Series("b", ["foo"], pl.Categorical),
            ]
        )
        df.write_parquet("/tmp/tmp.pq")
        with pl.StringCache():
>           out = pl.read_parquet("/tmp/tmp.pq").select(pl.col("b").value_counts())

tests/unit/io/test_lazy_parquet.py:207: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
polars/internals/dataframe/frame.py:5603: in select
    self.lazy().select(exprs).collect(no_optimization=True)._df
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <polars.LazyFrame object at 0x7F8EBD9BBDF0>

    def collect(
        self,
        *,
        type_coercion: bool = True,
        predicate_pushdown: bool = True,
        projection_pushdown: bool = True,
        simplify_expression: bool = True,
        no_optimization: bool = False,
        slice_pushdown: bool = True,
        common_subplan_elimination: bool = True,
        streaming: bool = False,
    ) -> pli.DataFrame:
        """
        Collect into a DataFrame.
    
        Note: use :func:`fetch` if you want to run your query on the first `n` rows
        only. This can be a huge time saver in debugging queries.
    
        Parameters
        ----------
        type_coercion
            Do type coercion optimization.
        predicate_pushdown
            Do predicate pushdown optimization.
        projection_pushdown
            Do projection pushdown optimization.
        simplify_expression
            Run simplify expressions optimization.
        no_optimization
            Turn off (certain) optimizations.
        slice_pushdown
            Slice pushdown optimization.
        common_subplan_elimination
            Will try to cache branching subplans that occur on self-joins or unions.
        streaming
            Run parts of the query in a streaming fashion (this is in an alpha state)
    
        Returns
        -------
        DataFrame
    
        Examples
        --------
        >>> df = pl.DataFrame(
        ...     {
        ...         "a": ["a", "b", "a", "b", "b", "c"],
        ...         "b": [1, 2, 3, 4, 5, 6],
        ...         "c": [6, 5, 4, 3, 2, 1],
        ...     }
        ... ).lazy()
        >>> df.groupby("a", maintain_order=True).agg(pl.all().sum()).collect()
        shape: (3, 3)
        ┌─────┬─────┬─────┐
        │ a   ┆ b   ┆ c   │
        │ --- ┆ --- ┆ --- │
        │ str ┆ i64 ┆ i64 │
        ╞═════╪═════╪═════╡
        │ a   ┆ 4   ┆ 10  │
        │ b   ┆ 11  ┆ 10  │
        │ c   ┆ 6   ┆ 1   │
        └─────┴─────┴─────┘
    
        """
        if no_optimization:
            predicate_pushdown = False
            projection_pushdown = False
            slice_pushdown = False
            common_subplan_elimination = False
    
        if streaming:
            common_subplan_elimination = False
    
        ldf = self._ldf.optimization_toggle(
            type_coercion,
            predicate_pushdown,
            projection_pushdown,
            simplify_expression,
            slice_pushdown,
            common_subplan_elimination,
            streaming,
        )
>       return pli.wrap_df(ldf.collect())
E       exceptions.NotFoundError: b
E       
E       > Error originated just after operation: '  DF ["name", "amount"]; PROJECT */2 COLUMNS; SELECTION: "None"
E       '
E       This operation could not be added to the plan.

polars/internals/lazyframe/frame.py:1147: NotFoundError
```

### Reproducible example

```python
None
```


### Expected behavior

Run tests successfully.

### Installed versions

~

--- a/py-polars/polars/internals/lazyframe/frame.py
+++ b/py-polars/polars/internals/lazyframe/frame.py
@@ -1148,7 +1148,7 @@ naive plan: (run LazyFrame.describe_optimized_plan() to see the optimized plan)
 
     def sink_parquet(
         self,
-        path: str,
+        path: str | Path,
         *,
         compression: str = "zstd",
         compression_level: int | None = None,
@@ -1248,7 +1248,7 @@ naive plan: (run LazyFrame.describe_optimized_plan() to see the optimized plan)
 
     def sink_ipc(
         self,
-        path: str,
+        path: str | Path,
         *,
         compression: str | None = "zstd",
         maintain_order: bool = True,

--- a/py-polars/polars/internals/lazyframe/frame.py
+++ b/py-polars/polars/internals/lazyframe/frame.py
@@ -1219,7 +1219,7 @@ naive plan: (run LazyFrame.describe_optimized_plan() to see the optimized plan)
         Examples
         --------
         >>> ldf = pl.scan_csv("/path/to/my_larger_than_ram_file.csv")  # doctest: +SKIP
-        >>> ldf.sink_parquet("/tmp/out.parquet")  # doctest: +SKIP
+        >>> ldf.sink_parquet("out.parquet")  # doctest: +SKIP
 
         """
         if no_optimization:
@@ -1294,7 +1294,7 @@ naive plan: (run LazyFrame.describe_optimized_plan() to see the optimized plan)
         Examples
         --------
         >>> ldf = pl.scan_csv("/path/to/my_larger_than_ram_file.csv")  # doctest: +SKIP
-        >>> ldf.sink_ipc("/tmp/out.arrow")  # doctest: +SKIP
+        >>> ldf.sink_ipc("out.arrow")  # doctest: +SKIP
 
         """
         if no_optimization:

--- a/py-polars/tests/unit/io/test_lazy_parquet.py
+++ b/py-polars/tests/unit/io/test_lazy_parquet.py
@@ -179,19 +179,29 @@ def test_parquet_statistics(
 
 @pytest.mark.xfail(sys.platform == "win32", reason="Does not work on Windows")
 def test_streaming_categorical() -> None:
-    pl.DataFrame(
+    df = pl.DataFrame(
         [
             pl.Series("name", ["Bob", "Alice", "Bob"], pl.Categorical),
             pl.Series("amount", [100, 200, 300]),
         ]
-    ).write_parquet("/tmp/tmp.pq")
-    with pl.StringCache():
-        assert pl.scan_parquet("/tmp/tmp.pq").groupby("name").agg(
-            pl.col("amount").sum()
-        ).collect().to_dict(False) == {
-            "name": ["Bob", "Alice"],
-            "amount": [400, 200],
-        }
+    )
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+        file_path = Path(temp_dir) / "categorical.parquet"
+        df.write_parquet(file_path)
+
+        with pl.StringCache():
+            result = (
+                pl.scan_parquet(file_path)
+                .groupby("name")
+                .agg(pl.col("amount").sum())
+                .collect()
+            )
+            expected = pl.DataFrame(
+                {"name": ["Bob", "Alice"], "amount": [400, 200]},
+                schema_overrides={"name": pl.Categorical},
+            )
+            assert_frame_equal(result, expected)
 
 
 @pytest.mark.xfail(sys.platform == "win32", reason="Does not work on Windows")
@@ -202,7 +212,11 @@ def test_parquet_struct_categorical() -> None:
             pl.Series("b", ["foo"], pl.Categorical),
         ]
     )
-    df.write_parquet("/tmp/tmp.pq")
-    with pl.StringCache():
-        out = pl.read_parquet("/tmp/tmp.pq").select(pl.col("b").value_counts())
-    assert out.to_dict(False) == {"b": [{"b": "foo", "counts": 1}]}
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+        file_path = Path(temp_dir) / "categorical.parquet"
+        df.write_parquet(file_path)
+
+        with pl.StringCache():
+            out = pl.read_parquet(file_path).select(pl.col("b").value_counts())
+        assert out.to_dict(False) == {"b": [{"b": "foo", "counts": 1}]}

--- a/py-polars/tests/unit/io/test_parquet.py
+++ b/py-polars/tests/unit/io/test_parquet.py
@@ -13,7 +13,7 @@ import pyarrow.parquet as pq
 import pytest
 
 import polars as pl
-from polars.testing import assert_frame_equal_local_categoricals
+from polars.testing import assert_frame_equal, assert_frame_equal_local_categoricals
 
 if TYPE_CHECKING:
     from polars.internals.type_aliases import ParquetCompression
@@ -350,34 +350,55 @@ def test_parquet_nested_dictionaries_6217() -> None:
 
 
 @pytest.mark.xfail(sys.platform == "win32", reason="Does not work on Windows")
-def test_sink_parquet_ipc(io_files_path: Path) -> None:
+def test_sink_parquet(io_files_path: Path) -> None:
     file = io_files_path / "small.parquet"
 
-    dst = "/tmp/test_sink.parquet"
-    pl.scan_parquet(file).sink_parquet(dst)
-    with pl.StringCache():
-        assert pl.read_parquet(dst).frame_equal(pl.read_parquet(file))
+    with tempfile.TemporaryDirectory() as temp_dir:
+        file_path = Path(temp_dir) / "sink.parquet"
 
-    dst = "/tmp/test_sink.ipc"
-    pl.scan_parquet(file).sink_ipc(dst)
-    with pl.StringCache():
-        assert pl.read_ipc(dst).frame_equal(pl.read_parquet(file))
+        df_scanned = pl.scan_parquet(file)
+        df_scanned.sink_parquet(file_path)
+
+        with pl.StringCache():
+            result = pl.read_parquet(file_path)
+            df_read = pl.read_parquet(file)
+            assert_frame_equal(result, df_read)
+
+
+@pytest.mark.xfail(sys.platform == "win32", reason="Does not work on Windows")
+def test_sink_ipc(io_files_path: Path) -> None:
+    file = io_files_path / "small.parquet"
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+        file_path = Path(temp_dir) / "sink.ipc"
+
+        df_scanned = pl.scan_parquet(file)
+        df_scanned.sink_ipc(file_path)
+
+        with pl.StringCache():
+            result = pl.read_ipc(file_path)
+            df_read = pl.read_parquet(file)
+            assert_frame_equal(result, df_read)
 
 
 @pytest.mark.xfail(sys.platform == "win32", reason="Does not work on Windows")
 def test_fetch_union() -> None:
-    pl.DataFrame({"a": [0, 1, 2], "b": [1, 2, 3]}).write_parquet(
-        "/tmp/df_fetch_1.parquet"
-    )
-    pl.DataFrame({"a": [3, 4, 5], "b": [4, 5, 6]}).write_parquet(
-        "/tmp/df_fetch_2.parquet"
-    )
+    df1 = pl.DataFrame({"a": [0, 1, 2], "b": [1, 2, 3]})
+    df2 = pl.DataFrame({"a": [3, 4, 5], "b": [4, 5, 6]})
 
-    assert pl.scan_parquet("/tmp/df_fetch_1.parquet").fetch(1).to_dict(False) == {
-        "a": [0],
-        "b": [1],
-    }
-    assert pl.scan_parquet("/tmp/df_fetch_*.parquet").fetch(1).to_dict(False) == {
-        "a": [0, 3],
-        "b": [1, 4],
-    }
+    with tempfile.TemporaryDirectory() as temp_dir:
+        file_path_1 = Path(temp_dir) / "df_fetch_1.parquet"
+        file_path_2 = Path(temp_dir) / "df_fetch_2.parquet"
+        file_path_glob = Path(temp_dir) / "df_fetch_*.parquet"
+
+        df1.write_parquet(file_path_1)
+        df2.write_parquet(file_path_2)
+
+        result_one = pl.scan_parquet(file_path_1).fetch(1)
+        result_glob = pl.scan_parquet(file_path_glob).fetch(1)
+
+    expected = pl.DataFrame({"a": [0], "b": [1]})
+    assert_frame_equal(result_one, expected)
+
+    expected = pl.DataFrame({"a": [0, 3], "b": [1, 4]})
+    assert_frame_equal(result_glob, expected)

